core/src/main/java/com/google/googlejavaformat/FormatterDiagnostic.java

```diff
@@ -35,6 +35,26 @@ public FormatterDiagnostic(String filename, int lineNumber, int column, String m
     this.message = message;
   }
 
+  /** Returns the name of the file that was processed. */
+  public String filename() {
+    return filename;
+  }
+
+  /** Returns the line number on which the error occurred. */
+  public int line() {
+    return lineNumber;
+  }
+
+  /** Returns the column number on which the error occurred. */
```

Comment:
nit: maybe "Returns the 0-indexed column number ..." since it's not obvious which to expect.